paymail_service_provider.go

```diff
@@ -172,6 +173,22 @@ func (p *PaymailDefaultServiceProvider) RecordTransaction(ctx context.Context,
 		return nil, err
 	}
 
+	if p2pTx.DecodedBeef == nil {
+		return &paymail.P2PTransactionPayload{
+			Note: p2pTx.MetaData.Note,
+			TxID: transaction.ID,
+		}, nil
+	}
+
+	if reflect.TypeOf(rts) == reflect.TypeOf(&externalIncomingTx{}) {
```

Comment:
Run it in a separate go routine. In the case of an error log, all the data needed to investigate what happened and manually save the transactions to the database by admin